### PR TITLE
feat: automatically refreshing folder structure on each commit

### PR DESCRIPTION
**Pull request by himanshu010**
fixes: #1320
fixes: #1310 
fixes: #1297 
**What kind of change does this PR introduce?**

Automating the refreshing of folder-structure.md.
**Did you add tests for your changes?**

**If relevant, did you update the documentation?**

**Summary**

**Does this PR introduce a breaking change?**

**Other information**

### CODE REVIEW

**Comment by himanshu010** [comment]
> This change should be done in the PR where you added the script. A separate PR is not required.
> This makes no sense as the script `createProjectFolderStructure.js` is not present here at all.

As I don't ask for reviews on the issue before sending the pr. Sending this to the same pr will create more confusion as it is solving a specific issue. Now reviewers can review them separately.

**Comment by snitin315** [comment]
> Sending this to the same pr will create more confusion 

Why??? it's just a single line.

More PR's ---> More confusion

**Comment by himanshu010** [comment]
> Please shift everything from other PRs into this PR and close them.

@rishabh3112 done 👍

File: scripts/createProjectFolderStructure.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
+const fs = require('fs');
+const path = require('path');
+const util = require('util');
+const directoryName = require('path').resolve(__dirname, '..');
+
+var logFile = fs.createWriteStream('folder-structure.md', { flags: 'w' });
+var logStdout = process.stdout;
```
Comment (snitin315) [inline scripts/createProjectFolderStructure.js]: somewhere `var` & somewhere `const`. please use `let / const` only.

File: package.json [hunk 2]
```diff
@@ -27,7 +27,7 @@
     "build": "node scripts/buildPackages.js",
     "prepsuite": "node scripts/prepareSuite.js",
     "clean:all": "rimraf node_modules packages/*/{node_modules}",
-    "commit": "git-cz",
+    "commit": "node scripts/createProjectFolderStructure.js &&  git add folder-structure.md && git-cz",
     "docs": "typedoc",
```
Comment (snitin315) [inline package.json]: LIMITATION - `we have to be present at root directory`

File: scripts/createProjectFolderStructure.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+console.log = function buildingStructure() {
+    logFile.write(util.format.apply(null, arguments) + '\n');
```
Comment (snitin315) [inline scripts/createProjectFolderStructure.js]: Please don't override built-in functions.

File: scripts/createProjectFolderStructure.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+        //ignored folders
+        const blacklist = {
+            '.git': true,
+            '.nyc_output': true,
+            coverage: true,
+            node_modules: true,
+            bin: true,
+            lib: true,
+        };
+        if (!blacklist[files[x]]) {
```
Comment (jamesgeorge007) [inline scripts/createProjectFolderStructure.js]: It seems a better approach would be to have `blacklist` as an arrray.

```js
      const blacklist = [ 
        '.git',
        '.nyc_output',
         'coverage',
         'node_modules',
         'bin',
         'lib',
      ];

      if (!blacklist.includes(files[x])) {
        //
      }
```

File: scripts/createProjectFolderStructure.js [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+                    break;
+                case 'smoketests':
+                    description = '## It contains minimal set of tests run on each build';
+                    break;
+                default:
+                    description = ' ';
+            }
+
+            var next = path.join(dir, files[x]);
+            if (fs.lstatSync(next).isDirectory() === true) {
```
Comment (jamesgeorge007) [inline scripts/createProjectFolderStructure.js]:
```suggestion
            if (fs.lstatSync(next).isDirectory()) {
```

File: scripts/createProjectFolderStructure.js (hunk 4 above)
Comment (himanshu010) [reply scripts/createProjectFolderStructure.js]: Yeah, this is better than creating an object

**Comment by himanshu010** [comment]
@jamesgeorge007 done 👍

**Comment by rishabh3112** [comment]
Sorry for bad communication, but I would like to discuss more about the benefits of this. I have shared views about it here https://github.com/webpack/webpack-cli/issues/1297#issuecomment-601102502.
For now I am closing this Pull request, we can definitely reopen it if we come out with affirmation about this approach.
Thanks for working on this!

**Pull request closed without merging**